.github/workflows/test.yml

```diff
@@ -59,8 +59,19 @@ jobs:
       - name: Build client
         run: cargo build -p holochain_client
 
+      - name: Lint
+        run: cargo clippy --all-targets --all-features -- -D warnings
```

Comment:
There were a lot of clippy violations, some of which took some thought to put right, better avoided early :)